docs/rfcs/003-consolidate-core-packages.md

```diff
@@ -0,0 +1,338 @@
+
+Also, as mentioned above, there's the added duty of doing the package "version dance" when we merge any new PRs to a package repository: publish the package update, update `package.json` in Atom.  It's very easy to forget to do this and not have community contributions included in the next Atom release!
+
+The more core packages live in `atom/atom`, the less work Atom maintainers have to do overall.
+
+## Explanation
+
+Many of Atom's core packages now live in the core `atom/atom` repository.  To the Atom user, this change will be imperceptible as these packages still show up in the list of Core Packages in the Settings View.  Users can still disable these packages if they want to replace their behavior with other packages.
+
+For maintainers and contributors, there will be less juggling of repositories and no more publishing of updates to these packages with `apm`:
+
+Contributors now clone and build `atom/atom` to work on improvements to core packages.  They will no longer have to use `apm link` in dev mode to test changes they make to packages in the repo's `packages` folder.  Core packages that aren't consolidated still have folders under `packages` with README.md files that point to the home repository for that package.
+
+When a contributor sends a PR to `atom/atom` that only affects files in a folder under `packages`, only the specs for the relevant package folders will be executed using Atom's CI scripts.  This means that a full Atom build will not be required when no Atom Core code is changed in a PR.  Package specs are also now run against all 3 OSes on Atom `master` and release builds.
+
+Atom maintainers no longer have to publish new versions to consolidated core packages and then edit `package.json` to bump the package version in a particular Atom release branch (Stable, Beta, or `master`).  When a PR against a consolidated core package in `atom/atom` is merged, no version number change is required and the changes will immediately be a part of the next release from that branch.
+
+## Drawbacks
+
+One possible drawback of this approach is that there might be some initial confusion where core Atom packages live, especially if some are consolidated into `atom/atom` and others still live in their own repositories.  We will manage this confusion by doing the following:
+
+- Include folders for _all_ core packages in the `packages` folder of the Atom repo and add README.md files to folders of those packages that still live in separate repos.  This will allow us to direct users to the proper home for packages that are not yet consolidated.
+
+- Archive the repositories for consolidated core packages, but only after migrating existing issues, merging or closing existing PRs, and updating the README.md to point to the new home of the package code.
```

Comment:
By the way, I'm pretty sure we could bulk-migrate open pull requests with the GraphQL API and some `git subtree` magic.

Reply:
Certainly could work, and might be easier than going through and trying to merge or close them all.  We'd probably lose existing review comments though, not sure if it's possible to recreate them all in a faithful way